libgnucash/engine/gnc-option-date.cpp

```diff
@@ -537,9 +519,9 @@ gnc_relative_date_to_time64(RelativeDatePeriod period)
             else if (reldate_is_next(period))
                 now.tm_mon += 3;
             if (gnc_relative_date_is_ending(period))
-                now.tm_mon += 2;
+                now.tm_mon += (offset == RelativeDateOffset::QUARTER && acct_per.tm_mday > 1 ? 3 : 2);
```

Comment:
I don't care much for the conditional referring to the previous case here. How about instead having 
```
    if (gnc_relative_date_is_ending(period))
        --delta;
    now.tm_mon -= delta;
}
    [[fallthrough]]
```
in the `QUARTER` case?

Reply:
Done.